src/types.tsx

```diff
@@ -665,6 +687,20 @@ export interface ScreenStackHeaderConfigProps extends ViewProps {
    * Callback which is executed when screen header is detached
    */
   onDetached?: () => void;
+  /**
+   * Callback which is executed when a bar button item is pressed.
+   * @param buttonId ID of the button that was pressed
+   */
+  onPressHeaderBarButtonItem?: (
+    e: NativeSyntheticEvent<{ buttonId: string }>,
+  ) => void;
+  /**
+   * Callback which is executed when a bar button item is pressed.
+   * @param buttonId ID of the button that was pressed
+   */
+  onPressHeaderBarButtonMenuItem?: (
+    e: NativeSyntheticEvent<{ menuId: string }>,
+  ) => void;
```

Comment:
These should be annotated as iOS only.

Reply:
https://github.com/software-mansion/react-native-screens/pull/2987/commits/b798b55190d3101e53f566d00aacbe5bfa759470